### PR TITLE
Turning on XML documentation

### PR DESCRIPTION
**Pull request by jtaubensee**
## Description
Addresses #26.

* Turned on XML documentation
* Added all basic documentation
  * Added empty tag `<summary></summary>` to all protected members that aren't designed to be extended by consumers of this library. Since these are easily searchable, we can always add this documentation. The main concern here is that I would like the warnings to be present in a build in case anyone adds any new members that aren't documented.
* Removed `ReceiverDisconnectedException`, since its only applicable to Event Hubs
* Made the following classes internal:
  * ServiceBusNamespaceConnection
  * ServiceBusConnection
  * AmqpSendReceiveLinkCreator
  * AmqpResponseMessage
  * AmqpRequestResponseLinkCreator
  * AmqpRequestMessage
  * AmqpLinkCreator
  * AmqpConnectionHelper
 
This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](./CONTRIBUTING.md).**
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR.
- [x] The pull request does not introduce breaking changes (unless a major version change occurs in the assembly and module).
- [x] If applicable, the public code is properly documented.
- [x] Pull request includes test coverage for the included changes.
- [x] The code builds without any errors.

### CODE REVIEW

**Comment by jtaubensee** [comment]
I currently see 239 errors, so this is definitely a work in progress.

**Comment by SeanFeldman** [comment]
Precisely why needs to be done early one 😃 
❤️  you @jtaubensee for getting the ball rolling on this one.

**Comment by jtaubensee** [comment]
What was done in this PR:
* Turned on XML documentation
* Treating warnings as *warnings*, since XML documentation is now enabled
* Added all basic documentation
  * Added empty tag `<summary></summary>` to all protected members that aren't designed to be extended by consumers of this library. Since these are easily searchable, we can always add this documentation. The main concern here is that I would like the warnings to be present in a build in case anyone adds any new members that aren't documented.
* Moved `Message.DeadLetterSource` to SystemProperties
* Removed `ReceiverDisconnectedException`, since its only applicable to Event Hubs
* Made the following classes internal:
  * ServiceBusNamespaceConnection
  * ServiceBusConnection
  * AmqpSendReceiveLinkCreator
  * AmqpResponseMessage
  * AmqpRequestResponseLinkCreator
  * AmqpRequestMessage
  * AmqpLinkCreator
  * AmqpConnectionHelper

**Comment by jtaubensee** [comment]
@nemakam / @vinaysurya - Please take a look, this should be good to go.

File: test/Microsoft.Azure.ServiceBus.UnitTests/Microsoft.Azure.ServiceBus.UnitTests.csproj [hunk 1]
```diff
@@ -3,7 +3,7 @@
   <PropertyGroup>
     <AssemblyTitle>Microsoft.Azure.ServiceBus.UnitTests</AssemblyTitle>
     <TargetFrameworks>netcoreapp1.0;net46</TargetFrameworks>
-    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <TreatWarningsAsErrors>false</TreatWarningsAsErrors>
```
Comment (SeanFeldman) [inline test/Microsoft.Azure.ServiceBus.UnitTests/Microsoft.Azure.ServiceBus.UnitTests.csproj]: Hmmm, in general, warnings should be treated as errors and exceptions should be excluded with pragmas :wink:

Comment (jtaubensee) [inline test/Microsoft.Azure.ServiceBus.UnitTests/Microsoft.Azure.ServiceBus.UnitTests.csproj]: Hey @SeanFeldman - I definitely agree with you, but I have one hesistation:

If someone were to create a 'spiked pr' as you call it, they would then have build errors if they didn't document all public memebers, which would prevent the CI from running. If they wanted to get rid of those warning they would need to have empty tags, at which point those errors wouldn't show up anymore.

Personally, I'd rather see the annoying warning, but continue the CI, and add something to our PR template, saying that there should not be any warnings during compilation.

Comment (jtaubensee) [inline test/Microsoft.Azure.ServiceBus.UnitTests/Microsoft.Azure.ServiceBus.UnitTests.csproj]: Ok, I was veto'd on that one... haha

File: src/Microsoft.Azure.ServiceBus/ISubscriptionClient.cs [hunk 2]
```diff
@@ -7,10 +7,19 @@ namespace Microsoft.Azure.ServiceBus
...
+    /// </summary>
     public interface ISubscriptionClient : IReceiverClient
     {
+        /// <summary>
+        /// Gets the path of the topic, for this subscription.
+        /// </summary>
         string TopicPath { get; }
 
+        /// <summary>
+        /// Gets the path of subscription.
```
Comment (nemakam) [inline src/Microsoft.Azure.ServiceBus/ISubscriptionClient.cs]: >path [](start = 21, length = 4)

name of subscription.

File: src/Microsoft.Azure.ServiceBus/Message.cs [hunk 3]
```diff
@@ -136,27 +171,58 @@ public TimeSpan TimeToLive
...
+        /// </summary>
+        public long Size { get; internal set; }
+
+        /// <summary>
+        /// Gets the user property bag, which can be used for custom message properties.
+        /// </summary>
         public IDictionary<string, object> UserProperties { get; internal set; }
 
+        /// <summary>
+        /// Gets the <see cref="SystemPropertiesCollection"/>, which is ised to store properties that are set by the system.
```
Comment (nemakam) [inline src/Microsoft.Azure.ServiceBus/Message.cs]: >ised [](start = 72, length = 4)

used

File: src/Microsoft.Azure.ServiceBus/Message.cs [hunk 4]
```diff
@@ -271,6 +344,7 @@ internal set
                 }
             }
 
+            /// <summary>Gets the unique number assigned to a message by Service Bus.</summary>
```
Comment (nemakam) [inline src/Microsoft.Azure.ServiceBus/Message.cs]: >unique number [](start = 34, length = 13)

for this entity

File: test/Microsoft.Azure.ServiceBus.UnitTests/Microsoft.Azure.ServiceBus.UnitTests.csproj (hunk 1 above)
Comment (SeanFeldman) [inline test/Microsoft.Azure.ServiceBus.UnitTests/Microsoft.Azure.ServiceBus.UnitTests.csproj]: What, I have insiders?! 😄

**Pull request merged**